CONTRIBUTING.md

```diff
@@ -0,0 +1,101 @@
+| Label | Description |
+| ----- | ----------- |
+| **[WIP]** | Work in Progress |
+| **[RFC]** | Request for Comments (aka Ready for Review) |
+| **[RDY]** | Ready |
+
+Please make sure you label and update your Pull Request accordingly. Alternatively, you can submit a **Draft Pull Request** as an alternative to the **[WIP]** label.
+
+
+#### CONTRIBUTING CODE: ####
+*NB: If you're using GitHub Desktop ignore the git steps and follow guidance from [GitHub Docs](https://docs.github.com/en/desktop/installing-and-configuring-github-desktop/overview/getting-started-with-github-desktop)*\
+**First Time**
+1. Ensure you have a GitHub account (https://github.com/signup/free)
+2. Fork CorsixTH\CorsixTH (https://github.com/CorsixTH/CorsixTH/fork)
+3. Ensure you have a git client.  ([GitHub Desktop](http://desktop.github.com) [Interactive] | [Git](https://git-scm.com/downloads) [Shell])
+4. Clone your fork to your computer
+	- ``git clone https://github.com/mygithubuser/CorsixTH.git``
+5. Add upstream remote
+	- ``git remote add upstream https://github.com/CorsixTH/CorsixTH.git``
+
+**Every Time**
+1. Sync your master branch with the CorisxTH repository's master branch
+	- ``git fetch upstream``
+	- ``git rebase upstream/master``
+2. Make sure no one is already working on the issue you want to work on.
+3. Tell other developers that you've started/will start working on this issue
+by posting a comment in its existing issue discussion or if there's no existing
+discussion for it then please open a new issue discussion for it and tell other
+devs here that your working on it.
+4. Create feature branch
+	- ``git branch myfeature upstream/master``
+5. Checkout your feature branch
+	- ``git checkout myfeature``
+6. Make your changes
+7. Unittest continuously, see README.txt in CorsixTH/Luatest for more info
+8. Review your changes
+	- ``git diff``
+	- ``git add`` | for each file
+	- ``git diff --check`` | Check there's no white spaces.
+	- ``git commit`` | Write an informative commit message and save
+9. Push your changes to your fork
+	- ``git push origin myfeature``
+10. Create a Pull Request -- GitHub will pick up your latest branch committed to automatically when accessing CorsixTH/CorsixTH.
+
+The developers will review and report any changes needed to your Pull Request so it could be accepted.
+
+#### Notes ####
+##### Multiple commits: #####
+If your feature is very big, break it into subprojects and do a separate commit
+for every independent operation.  This means doing step 10 several times as you
+go. You still only need to do one pull request at the end.
+
+##### Syncing with Upstream: #####
+If it takes a long time between when you start your feature and when you finish
+there might be other important changes other people are making to CorsixTH.  It
+is a good idea to make sure your code will still operate correctly with the
+latest changes.  To do this:
+- ``git fetch upstream``
```

Comment:
Again first `git checkout master`, since you sync `upstream/master` to your own `master`.
Also not `rebase` but `merge --ff-only` as before.

Then update your branch to the new master
``` shell
git checkout myfeature
git rebase master
```
and resolve any cpnflicts that may arise at this point.

Reply:
Can you write this into a suggestion of the correct use/order?